cobigen-plugins/cobigen-javaplugin-parent/cobigen-javaplugin/src/main/java/com/devonfw/cobigen/javaplugin/merger/JavaMerger.java

```diff
@@ -136,22 +136,61 @@ private String addGeneratedAnnotation(JavaClass baseClass, JavaClass parsedAnnot
     List<JavaConstructor> constructors = baseClass.getConstructors();
     List<JavaField> fields = baseClass.getFields();
     List<JavaMethod> methods = baseClass.getMethods();
+    // checks if the basefile has methods, constructors and fields present
+    // if so, then execute the rest of the code
     contentExist = constructors != null ? true : fields != null ? true : methods != null ? true : false;
 
     if (contentExist) {
-      baseClass.getSource().getImports().add("javax.annotation.Generated");
+      // checks if import is already present
+      List<String> imports = baseClass.getSource().getImports();
+      boolean importPresent = false;
+      for (String imp : imports) {
+        if (imp.equals("javax.annotation.Generated")) {
+          importPresent = true;
+          break;
+        }
+      }
+      if (!importPresent) {
+        baseClass.getSource().getImports().add("javax.annotation.Generated");
+      }
+      // Check if generated annotation already present, if yes then it does not add
       for (JavaConstructor constructor : constructors) {
-        List<JavaAnnotation> anno = constructor.getAnnotations();
-        ((AbstractBaseJavaEntity) constructor).setAnnotations(mergeAnnotation(anno, generatedAnnotation));
+        List<JavaAnnotation> annos = constructor.getAnnotations();
+        boolean annotationAlreadyPresent = false;
+        for (JavaAnnotation anno : annos) {
+          if (anno.getType().getFullyQualifiedName().equals("javax.annotation.Generated")) {
+            annotationAlreadyPresent = true;
```

Comment:
Is this boolean necessary? Why don't we put directly line 167 here instead?

Reply:
line 167 has to be executed only once in case if the annotation is not already present. Putting it here would mean that it is called  n(annos) times if the if condition holds and 0 times when the if clause is never true